Update troubleshooting guide for Spyder 6 (#419)
spyder-ide/spyder-docs: doc/troubleshooting/common-illnesses.rst

Completion/help not working
===========================

> To provide code completions, help and real-time analysis in the Editor, Spyder uses the Python LSP Server (PyLSP), an implementation of the Language Server Protocol specification used by VSCode, Atom and other popular editors/IDEs.

Reviewer: ```suggestion
To provide code completions, help and real-time analysis in the Editor, Spyder uses the Python LSP Server (PyLSP), an implementation of the Language Server Protocol specification used by other popular editors/IDEs (e.g. Neovim).
```

Our server is not used too much these days and the Atom reference is outdated now.

Author: The original semantics left it ambiguous, but the actual intended meaning was that "used by" referred to "the Language Server specification" immediately preceding it, not the specific PyLSP implementation mentioned earlier in the sentence.

To clarify this, we can instead say:

```suggestion
To provide code completions, help and real-time analysis in the Editor, Spyder uses the Python LSP Server (PyLSP), an implementation of the same Language Server Protocol specification supported by VSCode, Neovim and other popular editors/IDEs.
```

Plugin does not work at all
~~~~~~~~~~~~~~~~~~~~~~~~~~~

> If you have installed a Spyder plugin, but you can't see it, go to the :guilabel:`Panes` submenu of the :guilabel:`View` menu and select the plugin's name, which should make it visible if it contains a standalone pane.

Reviewer: ```suggestion
If you have installed a Spyder plugin, but you can't see it, go to the :guilabel:`Panes` submenu of the :guilabel:`Window` menu and select the plugin's name, which should make it visible if it contains a standalone pane.
```

That's the menu's name in 6.1+

Author: Huh, I wonder who's idea was it to change that 😅

> If you don't see the plugin there, select the :guilabel:`Plugins` pane under Spyder's :guilabel:`Preferences` and see if it is listed there.

Reviewer: ```suggestion
If you don't see the plugin there, select the :guilabel:`Plugins` section under Spyder's :guilabel:`Preferences` and see if it is listed there.
```

Uniformize the term used for that in Preferences (that's the one you used in PR #418).

.. image:: /images/common-illnesses/common-illnesses-plugins.png
   :alt: Dependencies dialog showing Unittest plugin

> If the plugin with the problem is not listed in the :guilabel:`Plugins` Preferences pane, check that you installed it in the same environment as Spyder.

Reviewer: ```suggestion
If the plugin with the problem is not listed in the :guilabel:`Plugins` Preferences section, check that you installed it in the same environment as Spyder.
```
